### ISSUE TITLE
Precipitation downscaling/bias correction interface

### ISSUE THREAD

**Issue by milankl** [SpeedyWeather/SpeedyWeather.jl#370]
After a chat with @p-hss who is interested in online ML-based downscaling/bias-correction for precipitation this is an issue to summarize ideas

- As downscaling is inherently a 2D+ method (information from neighbouring grid cells needs to be available) a `precipitation_correction!` (working title) function call would need to happen after the loop over all columns in `src/physics/tendencies.jl`
- This method may not easily generalise to the variety of grids we support. For the beginning methods could be defined via dispatch to only for the `FullClenshawGrid`(<: regular lat-lon) with nothing-fallbacks for other grids.
- At the moment the precipitation parameterizations accumulate their tendency in one tendency array (per variable) for all parameterizations. However, the mm of rain are stored in `diagnostic_variables.surface.precip_...` for diagnostics and output (for which it is set back to zero), but there they are accumulated in time. For the required ML input we'd need to make sure to store in one surface field that has condensation+convection since the last time step.
- To make the ML downscaling/bias correction an *online* scheme, we'd then need to reobtain tendencies at every vertical level that are also dynamically consistent. This could work via normalization: Say level 1 produces 1mm, level 2 produces 2mm, so 3mm in total, but correction says it should only be 2mm, then scale tendencies by 2/3. But recompute tendencies like condensation heating from it. Maybe this points to a 1D parameterization pass, then 2D then 1D again.

**Comment by bgroenks96** [comment]
Maybe @p-hss can correct me if I'm wrong, but from recent discussions, I think we have decided not to do this?

Rather, we would want to apply bias correction directly to the prognostic state variables (if at all) and implement downscaling such that it preserves the coarse scale quantities.

**Comment by p-hss** [comment]
Yep, I agree, the ideas outlined above are more about an online correction scheme for the precipitation parameterization than the online bias correction full prognostic state or downscaling that we've been discussing recently.

**Comment by bgroenks96** [comment]
@milankl Could we close this then?
